### ISSUE TITLE
Add support for scripting on Linux

### ISSUE THREAD

**Issue by aryanbaburajan** [DucktapeEngine/Ducktape#107]
As of now, Ducktape uses the `Window.h` library for loading dlls for Native Scripting. This means that it would not work on Linux, and it'd be great if someone added Linux support using the builtin linux libraries (that contain `dlopen()`, etc), and used the platform based `#ifdef` guards to use the Linux or Windows library appropriately like:
```
#if defined(WINDOWS_MACRO)
#include <windowsLibrary>
#elif defined(LINUX_MACRO)
#include <linuxLibrary>
#endif
```
and did the same for the rest of the Dll loading functions.

The platform dependant functions are [here](https://github.com/DucktapeEngine/Ducktape/blob/main/Engine/Core/Entity.h#L69) and [here](https://github.com/DucktapeEngine/Ducktape/blob/main/Engine/Core/Scene.cpp#L44).

**Comment by SuKiN-a** [comment]
i'll take a stab at this

**Comment by aryanbaburajan** [comment]
Fixed in #109.
